src/features/transfer-ownership/TransferOwnershipForm/TransferOwnershipForm.test.tsx

```diff
@@ -0,0 +1,465 @@
+
+			expect(input).toBeInTheDocument();
+
+			await act(async () => {
+				fireEvent.input(input, { target: { value: mock.connectedAccount } });
+			});
+
+			expect(onConfirmInputButton).not.toHaveAttribute('aria-disabled');
+
+			fireEvent.click(onConfirmInputButton);
+
+			await screen.findByText(errorMessage);
+		});
+
+		test('should display error message when connected account is not domain owner(0xxx)', async () => {
+			// edit mock.domain.owner
+			const mockOwner = '0xxx';
+
+			renderComponent(mockOwner);
+
+			// input error message
+			const errorMessage = 'You are not the owner of this domain';
+
+			expect(screen.queryByText(errorMessage)).not.toBeInTheDocument();
+
+			const onConfirmInputButton = await screen.findByRole('button', {
+				name: /transfer/i,
+			});
+
+			expect(onConfirmInputButton).toBeInTheDocument();
+			expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+
+			const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+			expect(input).toBeInTheDocument();
+
+			await act(async () => {
+				fireEvent.input(input, { target: { value: mock.validInputAddress } });
+			});
+
+			expect(onConfirmInputButton).not.toHaveAttribute('aria-disabled');
+
+			fireEvent.click(onConfirmInputButton);
+
+			await screen.findByText(errorMessage);
+		});
+	});
+
+	describe('Confirm Step', () => {
+		test('secondary button should call onClose when clicked', async () => {
+			testSetup();
+			onSubmitValidDetails();
+
+			await screen.findByText(confirmScreenText);
+
+			const onCancelButton = await screen.findByText(/cancel/i);
+
+			expect(onCancelButton).toBeInTheDocument();
+
+			fireEvent.click(onCancelButton);
+
+			expect(mockOnClose).toBeCalled();
+		});
+
+		test('primary button should call transferDomainOwnership when clicked', async () => {
+			testSetup();
+			onSubmitValidDetails();
+
+			await screen.findByText(confirmScreenText);
+
+			const onConfirmButton = await screen.findByText(/confirm/i);
+
+			expect(onConfirmButton).toBeInTheDocument();
+
+			fireEvent.click(onConfirmButton);
+
+			expect(mockTranferDomainOwnership).toBeCalledTimes(1);
+		});
+	});
+
+	describe('Transaction Approval Step (signature)', () => {
+		test('should navigate back to confirm step if signature is rejected', async () => {
+			mockTranferDomainOwnership.mockRejectedValue(undefined);
+			testSetup();
+			onSubmitValidDetails();
+
+			await screen.findByText(confirmScreenText);
+
+			const onConfirmTransactionButton = await screen.findByText(/confirm/i);
+
+			expect(onConfirmTransactionButton).toBeInTheDocument();
+
+			fireEvent.click(onConfirmTransactionButton);
+
+			expect(onConfirmTransactionButton).not.toBeInTheDocument();
+
+			screen.getByText('Please accept wallet transaction..');
+
+			expect(mockTranferDomainOwnership).toBeCalledTimes(1);
+
+			await screen.findByText(/confirm/i);
+		});
+
+		test('should handle rejected signature', async () => {
```

Comment:
Are these two tests testing the same thing?

Reply:
Merged the two.